### PR TITLE
refactor: MIgrate NRIC validator to TypeScript

### PR DESCRIPTION
**Pull request by liangyuanruo**
## Problem

Addresses #7 

## Solution

**Improvements**:

- Migrates NRIC validation to Typescript
- Extracted the singleAnswerValidator factory out for code neatness

**Fixes**:

- Added missing `isVisible` property to Jasmine test responses, as all `ProcessedFieldResponse` responses should include said property.

## Tests
Manually tested both valid, invalid and hidden fields.

### CODE REVIEW

File: src/app/utils/field-validation/validators/nricValidator.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type NricValidator = ResponseValidator<ISingleAnswerResponse>
+type NricValidatorConstructor = () => NricValidator
+
+const nricValidator: NricValidator = (response) => {
+  return isNricValid(response.answer)
+    ? right(response)
+    : left(`NricValidator:\tanswer is not a valid NRIC`)
```
Comment (tshuli) [inline src/app/utils/field-validation/validators/nricValidator.ts:17]: not sure if I'm missing something but did you mean 
``: left(`NricValidator:\t ${response.answer} is not a valid NRIC`)``

**Review by tshuli — APPROVED** [review]
lgtm with one minor clarification

**Review by karrui — APPROVED** [review]
lgtm

File: src/app/utils/field-validation/validators/nricValidator.ts (hunk 1 above)
Comment (karrui) [reply src/app/utils/field-validation/validators/nricValidator.ts:17]: both works la, one is more dynamic one is not. After the injection thingamagic I think having a generic error message is fine

Comment (liangyuanruo) [reply src/app/utils/field-validation/validators/nricValidator.ts:17]: Actually, it is our policy to never log user-submitted answers for privacy reasons. FYI @karrui @mantariksh @tshuli

**Pull request merged**